fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingThrottle.java

```diff
@@ -72,95 +71,184 @@ public class IndexingThrottle {
             FDBError.COMMIT_READ_INCOMPLETE.code(),
             FDBError.TRANSACTION_TOO_LARGE.code()));
 
-    /**
-     * The number of successful transactions in a row as called by {@link #throttledRunAsync(Function, BiFunction, Function, List)}.
-     */
-    private int successCount = 0;
+    private class Booker {
+        /**
+         * Keep track of success/failures and adjust transaction's canned records limit when needed.
+         * Note that when duringRangesIteration=true,  a single thread processing is assumed.
```

Comment:
The throttle (via `buildCommitRetryAsync`) is used in two ways: 
1. Iterating range (or ranges) - which is done in a single thread by definition.
2. Endpoint operations (legacy code + insert known ranges)  - which might be done in parallel. Since these operations are typically done for a single item, they should not affect the Booker's counters. 
This duality might be cleaned if/when removing the old `IndexingByRecords` module.